scripts/deploy/troubleshooting/get-all-logs

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+function usage() {
+  echo "Usage: $0 <namespace>"
+  echo
+  echo "This script saves logs from all pods within the specified namespace into local files and then creates a tar.gz archive of these logs."
+  echo
+  echo "Arguments:"
+  echo "  namespace   - The namespace from which to retrieve and archive pod logs."
+  echo
+  echo "Example: $0 mynamespace"
+  exit 1
+}
+
+if [ "$#" -ne 1 ]; then
+  usage
+fi
+
+NAMESPACE=$1
+
+mkdir -p "$NAMESPACE-logs"
+if [ $? -ne 0 ]; then
+  echo "Failed to create directory for logs."
+  exit 1
+fi
+
+# Get a list of all pod names in the namespace
+PODS=$(kubectl -n $NAMESPACE get pods -o jsonpath="{.items[*].metadata.name}")
+
+if [ -z "$PODS" ]; then
+  echo "No Pods found in Namespace $NAMESPACE."
+  rm -rf "$NAMESPACE-logs"
```

Comment:
Instead of creating a directory using the `$NAMESPACE`, let's just use something like `/tmp/cht-eks-logs` or something.  We can create it and abandon it and let `tmp` do it's thing.  

Otherwise, accepting a user input variable and then running `rm -rf` with it is dangerous.  A slightly forced, but real enough, demo below:

```
$ mkdir really-important-stuff-omg-dont-delete-logs

$ file really-important-stuff-omg-dont-delete-logs
really-important-stuff-omg-dont-delete-logs: directory

$ ./troubleshooting/get-all-logs really-important-stuff-omg-dont-delete
No Pods found in Namespace really-important-stuff-omg-dont-delete.

$ file really-important-stuff-omg-dont-delete-logs                     
really-important-stuff-omg-dont-delete-logs: cannot open `really-important-stuff-omg-dont-delete-logs' (No such file or directory)
```

```suggestion
```

Reply:
handled via mktemp.